### DISCUSSION TITLE
Image skeleton or blur placeholder

### DISCUSSION THREAD

**Discussion by lampy255** [outline/outline#7139] [Ideas]
Would we be interested in implementing some kind of skeleton or blur placeholder for loading images?
The current user experience is that whilst a document with many images is loading, the document layout jumps and changes until everything has loaded.
It could be nice to have a placeholder that takes each images place until they have loaded?

Examples:
https://www.npmjs.com/package/react-native-skeleton-placeholder
https://github.com/mrousavy/react-native-blurhash

**Comment by tommoor** [comment]
Mostly it's there already, the only "gotcha" is that the image dimensions for the placeholder node are only stored when the image has been resized – you can test this by resizing an image smaller and bigger again then reload the page and you should see that the space is retained while loading.

TLDR: The fix is to store the dimensions for all images

**Reply by lampy255** [reply]
Ok I see. I guess I'm talking more about a visual indication to the user that content is being loaded.
But I do appreciate that going down the blur effect road could quickly become complex and messy.
